src/main/java/tasks/Event.java

```diff
@@ -0,0 +1,83 @@
+package main.java.tasks;
+import main.java.exceptions.InvalidDescriptionException;
+import main.java.exceptions.InvalidTimeException;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents an event, consisting of a description and a time.
+ * Throws InvalidDescriptionException or InvalidTimeException if
+ * the description or time provided is blank.
+ */
+public class Event extends Task {
+
+    private final String time;
+    private LocalDateTime formattedDateTime;
+    private LocalDate formattedDate;
+
+    public Event(String description, String time) throws InvalidDescriptionException, InvalidTimeException {
+        super(description);
+        if (description.isBlank()) {
+            throw new InvalidDescriptionException(
+                    "Hey! Event description shouldn't be blank.");
+        } else if (time.isBlank()) {
+            throw new InvalidTimeException(
+                    "Do try again by adding a time where the event takes place.");
+        } else {
+            this.time = time;
+            formatTime();
+        }
+    }
+
+    public Event(String description, String time, boolean isDone) {
+        super(description, isDone);
+        this.time = time;
+        formatTime();
+    }
+
+    private void formatTime() {
+        try {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+            this.formattedDateTime = LocalDateTime.parse(this.time, formatter);
+        } catch (DateTimeParseException e1) {
+            this.formattedDateTime = null;
+            try {
+                this.formattedDate = LocalDate.parse(this.time);
+            } catch (DateTimeParseException e2) {
```

Comment:
Maybe shall we try to avoid naming variables using number index(e2) here? But indeed as it has a very small scope, thus I think the naming may also be okay.